### DISCUSSION TITLE
mapping du champ last_update - labo 1: quarkus-and-sakila

### DISCUSSION THREAD

**Discussion by Greywind299792458** [amt-classroom/amt-classroom.github.io#2] [Q&A] [answered]
Bonjour @bchapuis nous avons une question au sujet du laboratoire, redites moi si c'est le bon endroit pour la poser et si ce n'est pas le cas je la déménage volontiers à l'endroit adapté !

Nous avons constaté que toutes les entités dans la base de données avait un attribut "last_update", qui semblait être automatiquement géré par la DB au moment d'une action sur un enregistrement pour inscrire la date/heure courante.

Nous hésitions sur la façon de mapper cet attribut. Si nous le faisions de manière classique, nos tests unitaires incluant des insertions ne passaient pas (il attendait une valeur pour ce champ du coup) ! Nous avions donc deux idées : 
- ne pas mapper cet attribut en considérant que celui ci est surtout destiné à la DB
- (variante que nous avons choisi pour le moment) -> mapper cet attribut et utiliser les annotations suivantes qui semblent indiquer que ce champ sera auto généré
```java
@UpdateTimestamp
@Temporal(TemporalType.TIMESTAMP)
```
Partons nous sur la bonne piste ? 

Cordialement,

Elliot et Alexis

**Comment by bchapuis** [comment, marked as answer]
Merci pour votre question, c'est le bon endroit pour poser des questions. Il est préférable de ne pas mapper cet attribut, car il est entièrement géré par la base de données grâce à des triggers. J'avais brièvement évoqué ce point oralement en classe, mais nous clarifierons cela par écrit dans l'intitulé pour les éditions futures du laboratoire.

**Reply by Greywind299792458** [reply]
merci beaucoup pour votre réponse !
